In [1]:
import pandas as pd

In [2]:
dataset=pd.read_csv("preprocessed_data.csv")

In [3]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed  265000.0  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed  265000.0  

[215 rows x 15 columns]

In [4]:
dataset.isnull().sum()

sl_no             0
gender            0
ssc_p             0
ssc_b             0
hsc_p             0
hsc_b             0
hsc_s             0
degree_p          0
degree_t          0
workex            0
etest_p           0
specialisation    0
mba_p             0
status            0
salary            0
dtype: int64

In [5]:
dataset.cov(numeric_only=True) 

sl_no         ssc_p          hsc_p      degree_p  \
sl_no       3870.000000    -52.641355     -58.106028    -40.413645   
ssc_p        -52.641355    117.228377      60.348373     42.897137   
hsc_p        -58.106028     60.348373     118.755706     34.819820   
degree_p     -40.413645     42.897137      34.819820     54.151103   
etest_p       52.556168     37.659225      35.461678     21.929469   
mba_p          8.102336     24.535952      22.555846     17.272020   
salary    230051.401869  92046.789176  104795.266681  30714.199304   

                etest_p         mba_p        salary  
sl_no         52.556168      8.102336  2.300514e+05  
ssc_p         37.659225     24.535952  9.204679e+04  
hsc_p         35.461678     22.555846  1.047953e+05  
degree_p      21.929469     17.272020  3.071420e+04  
etest_p      176.251018     16.886973  1.757669e+05  
mba_p         16.886973     34.028376  7.104318e+04  
salary    175766.850250  71043.180613  6.120326e+09

#CoVarience 
1.The covarience between degree_p and etest_p is 21.929469. this is 24 postive covarience and it is 24 times differs from the etest_p marks.
2.The covarience between etest_p and mba_p is 16.886973. In this 16 postive covarience and it is 16 times differs from mba_p marks 

In [6]:
dataset.corr(numeric_only=True)

sl_no     ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
sl_no     1.000000 -0.078155 -0.085711 -0.088281  0.063636  0.022327  0.047270
ssc_p    -0.078155  1.000000  0.511472  0.538404  0.261993  0.388478  0.108669
hsc_p    -0.085711  0.511472  1.000000  0.434206  0.245113  0.354823  0.122921
degree_p -0.088281  0.538404  0.434206  1.000000  0.224470  0.402364  0.053352
etest_p   0.063636  0.261993  0.245113  0.224470  1.000000  0.218055  0.169233
mba_p     0.022327  0.388478  0.354823  0.402364  0.218055  1.000000  0.155673
salary    0.047270  0.108669  0.122921  0.053352  0.169233  0.155673  1.000000

#Correlation 
1.Correlation between mba_p and Salary is 0.155673. it is nearly positive correlation but baseed on the mark is zero correlation and it in neutral mode.